docs/react-native-gradle-plugin.md

```diff
@@ -0,0 +1,187 @@
+The entrypoint file is needed as the plugin needs to invoke the CLI for bundling and creating your app.
+
+If you're in a monorepo or using a different package manager, you can use adjust `cliFile` to your setup.
+You can customize it as follows:
+
+```groovy
+cliFile = file("../node_modules/react-native/cli.js")
+```
+
+### `debuggableVariants`
+
+This is the list of variant that are debuggable (see [using variants](#using-variants) for more context on variants).
+
+By default the plugin is considering as `debuggableVariants` only `debug`, while `release` is not. If you have other
+variants (like `staging`, `lite`, etc.) you'll need to adjust this accordingly.
+
+Variants that are listed as `debuggableVariants` will not come with a shipped bundle, so you'll need Metro to run them.
+
+You can customize it as follows:
+
+```groovy
+debuggableVariants = ["liteDebug", "prodDebug"]
+```
+
+### `nodeExecutableAndArgs`
+
+This is the list of node command and arguments that should be invoked for all the scripts. By default is just `[node]` but can be customized to add extra flags as follows:
+
+```groovy
+nodeExecutableAndArgs = ["node"]
+```
+
+### `bundleCommand`
+
+This is the name of the `bundle` command to be invoked when creating the bundle for your app. That's useful if you're using [RAM Bundles](/docs/ram-bundles-inline-requires). By default is just `bundle` but can be customized to add extra flags as follows:
+
+```groovy
+bundleCommand = "ram-bundle"
+```
+
+### `bundleConfig`
+
+This is the path to a configuration file that will be passed to `bundle --config <file>` if provided. Default is empty (no config file will be probided). More information on bundling config files can be found [on the CLI documentation](https://github.com/react-native-community/cli/blob/main/docs/commands.md#bundle). Can be customized as follow:
+
+```groovy
+bundleConfig = file(../rn-cli.config.js)
+```
+
+### `bundleAssetName`
+
+This is the name of the bundle file that should be generated. Default is `index.android.bundle`. Can be customized as follow:
+
+```groovy
+bundleAssetName = "MyApplication.android.bundle"
+```
+
+### `entryFile`
+
+The entry file used for bundle generation. The default is to search for `index.android.js` or `index.js`. Can be customized as follow:
+
+```groovy
+entryFile = file("../js/MyApplication.android.js")
+```
+
+### `extraPackagerArgs`
+
+A list of extra flags that will be passed to the `bundle` command. The list of available flags is in [the CLI documentation](https://github.com/react-native-community/cli/blob/main/docs/commands.md#bundle). Default is empty. Can be customized as follows:
+
+```groovy
+extraPackagerArgs = []
+```
+
+### `hermesCommand`
+
+The path to the `hermesc` command (the Hermes Compiler). React Native comes with a version of the Hermes compiler bundled it with, so you generally won't be needing to customize this, by default the plugin will use the correct compiler for your system.
```

Comment:
```suggestion
The path to the `hermesc` command (the Hermes Compiler). React Native comes with a version of the Hermes compiler bundled with it, so you generally won't be needing to customize this. The plugin will use the correct compiler for your system by default.
```

If the plugin uses the right compiler, why do we have this key? Are there setups that requires its customization?